lprint.c

```diff
@@ -527,7 +527,7 @@ system_cb(
       else if (!strcmp(valptr, "web-security") || !strncmp(valptr, "web-security,", 13))
         soptions |= PAPPL_SOPTIONS_WEB_SECURITY;
       else if (!strcmp(valptr, "no-tls") || !strncmp(valptr, "no-tls,", 7))
-        soptions |= PAPPL_SOPTIONS_NO_TLS;
+        soptions &= (pappl_soptions_t)~PAPPL_SOPTIONS_WEB_TLS;
```

Comment:
So you need to remove/mask WEB_TLS and add NO_TLS, since the "no-tls" option disables TLS entirely, not just the TLS certificate web page.